packages/frontend/src/contexts/unirep.js

```diff
@@ -5,10 +5,12 @@ export default class Unirep {
   allSignUps = []
   signUpsByUserId = new Map()
   signUpsByAttesterId = new Map()
```

Comment:
With your `ingestAttestations()` as example, now I see how I should be creating a single `signUpsById` mapping here and be able to use it for both attesters and users. Sound right?

Reply:
You'll still need to track signups by attester, but it doesn't need to be in the global context. Instead it could be in the component only. e.g. the context has a function for loading the signups for an attester, then that value is returned to the component, and the component puts the array in a state variable.

The array of users that have signed up for an attester is pretty specific data that probably isn't needed in many places.